Кузнецов. 4 спринт. Проектная работа.

DimoonAndDev opened the pull request:
Добрый день! 

Приложение работает. Будто бы все понятно, но без детальной инструкции сам бы так делал не сделал. 
P.S. Второй Initial коммит сделан, так как в первом не добавил новые классы, которые нужно было создать по заданию. Прокликал этап комментария, потому получился второй Initial.

sbuvaka-fhl committed:
Add dev branch for Y.Practicum

DimoonAndDev committed:
Initial

DimoonAndDev committed:
Initial

faritowich commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/ListDiffInterface.java:
```diff
@@ -0,0 +1,11 @@
+import ru.yandex.practicum.contacts.presentation.sort.SortTypeUI;
+
+public interface ListDiffInterface<T> {
+    <T> boolean theSameAs(T object);
```
Дженерик тут не нужен, он объявлен на уровне интерфейса

DimoonAndDev replied:
done

faritowich commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/ListDiffInterface.java:
```diff
@@ -0,0 +1,11 @@
+import ru.yandex.practicum.contacts.presentation.sort.SortTypeUI;
+
+public interface ListDiffInterface<T> {
+    <T> boolean theSameAs(T object);
```
Лучше использовать такой нейминг, чтобы по названию было понятно, что именно параметр означает. Например, anotherItem вместо object

DimoonAndDev replied:
done

DimoonAndDev committed:
1.Critical! Добавлен метод getChangePayload в BaseListDiffCallback.
2. Critical! Исправлена правльная реализация метода theSameAs для классов  UI.
3. У метода theSameAs убрано объявление дженериком, тк интерфейс с дженериком.
4. Переименован параметр object для методов theSameAs.
